src/RobotInterface/YarpImplementation/include/BipedalLocomotion/RobotInterface/YarpSensorBridgeImpl.h

```diff
@@ -0,0 +1,1695 @@
+
+    std::unordered_map<std::string, yarp::dev::IGenericSensor*> wholeBodyAnalogIMUInterface; /** < map  of IMU sensors attached through generic sensor interfaces */
+    std::unordered_map<std::string, yarp::dev::IGenericSensor*> wholeBodyCartesianWrenchInterface; /** < map  of cartesian wrench streams attached through generic sensor interfaces */
+    std::unordered_map<std::string, yarp::dev::IAnalogSensor*> wholeBodyAnalogSixAxisFTSensorsInterface; /** < map  of six axis force torque sensors attached through analog sensor interfaces */
+    std::unordered_map<std::string, yarp::dev::IFrameGrabberImage*> wholeBodyFrameGrabberInterface; /** < map of cameras attached through frame grabber interfaces */
+    std::unordered_map<std::string, yarp::dev::IRGBDSensor*> wholeBodyRGBDInterface; /** < map of cameras attached through RGBD interfaces */
+
+    struct
+    {
+        std::unordered_map<std::string, size_t> accelerometer;
+        std::unordered_map<std::string, size_t> gyroscopes;
+        std::unordered_map<std::string, size_t> magnetometers;
+        std::unordered_map<std::string, size_t> orientationSensors;
+        std::unordered_map<std::string, size_t> sixAxisFTSensors;
+    }masSensorIndexMaps;
+
+    /**
+     * Struct holding measurements polled from remapped remote control board interfaces
+     */
+    struct
+    {
+        yarp::sig::Vector remappedJointIndices;
+        yarp::sig::Vector jointPositions;
+        yarp::sig::Vector jointVelocities;
+        double receivedTimeInSeconds;
+    } controlBoardRemapperMeasures;
+
+    std::unordered_map<std::string, StampedYARPVector> wholeBodyIMUMeasures; /** < map holding analog IMU sensor measurements */
+    std::unordered_map<std::string, StampedYARPVector> wholeBodyFTMeasures; /** < map holding six axis force torque measures */
+    std::unordered_map<std::string, StampedYARPVector> wholeBodyInertialMeasures; /** < map holding three axis inertial sensor measures */
+    std::unordered_map<std::string, StampedYARPVector> wholeBodyCartesianWrenchMeasures; /** < map holding cartesian wrench measures */
+    std::unordered_map<std::string, StampedYARPImage> wholeBodyCameraRGBImages; /** < map holding images **/
+    std::unordered_map<std::string, StampedYARPImage> wholeBodyCameraDepthImages; /** < map holding images **/
+
+    const int nrChannelsInYARPGenericIMUSensor{12};
+    const int nrChannelsInYARPGenericCartesianWrench{6};
+    const int nrChannelsInYARPAnalogSixAxisFTSensor{6};
+
+    std::vector<std::string> failedSensorReads;
+    SensorBridgeMetaData metaData; /**< struct holding meta data **/
+    bool bridgeInitialized{false}; /**< flag set to true if the bridge is successfully initialized */
+    bool driversAttached{false}; /**< flag set to true if the bridge is successfully attached to required device drivers */
+    bool sensorDryRunEnabled{false}; /**< flag to enable running a test stream of sensor interfaces after attaching to the device drivers */
```

Comment:
This flag is unused. I am not running a dry run read sensor anywhere!

Reply:
Removed in https://github.com/dic-iit/bipedal-locomotion-framework/pull/106/commits/9e6f3399f5c778a35bf36045fe8fa4cdc21b8d36